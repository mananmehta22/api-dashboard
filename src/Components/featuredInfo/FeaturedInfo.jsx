import React, {useState, useEffect} from "react";
import axios from "axios";
import "./featuredInfo.css";

export default function FeaturedInfo() {
    const urlUpcoming = 'https://api.spacexdata.com/v5/launches/upcoming';
    const urlPast = 'https://api.spacexdata.com/v5/launches/past';
    const Total = 'https://api.spacexdata.com/v5/launches';

    const [rocket, setRocket] = useState(null)
    const [rocket1, setRocket1] = useState(null)
    const [rocket2, setRocket2] = useState(null)

    useEffect(() => {
      axios.get(urlUpcoming)
            .then(response => {
                console.log(response)
                setRocket(response.data)
            })
            .catch(err => {
              console.log(err)
            })
    })

    useEffect(() => {
      axios.get(urlPast)
            .then(response => {
                console.log(response)
                setRocket1(response.data)
            })
            .catch(err => {
              console.log(err)
            })
    })

    useEffect(() => {
      axios.get(Total)
            .then(response => {
                console.log(response)
                setRocket2(response.data)
            })
            .catch(err => {
              console.log(err)
            })
    })
    
    return (
    <div className="featured">
      <div className="featuredItem">
        <span className="featuredTitle"></span>
        <div className="featuredLaunchContainer">
          <span className="featuredUpcoming">{rocket.length}</span>
        </div>
        <span className="featuredSub">Upcoming</span>
      </div>
      <div className="featuredItem">
        <span className="featuredTitle"></span>
        <div className="featuredLaunchContainer">
          <span className="featuredUpcoming">{rocket1.length}</span>
        </div>
        <span className="featuredSub">Past</span>
      </div>
      <div className="featuredItem">
        <span className="featuredTitle"></span>
        <div className="featuredLaunchContainer">
          <span className="featuredUpcoming">{rocket2.length}</span>
        </div>
        <span className="featuredSub">Total</span>
      </div>
    </div>
  );
}
