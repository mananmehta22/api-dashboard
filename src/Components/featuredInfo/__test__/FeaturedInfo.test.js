import { React } from 'react';
import { MockedProvider } from '@apollo/react-testing';
import { FeaturedInfo, FETCH_API } from '../FeaturedInfo';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
const mocks =
    {
request: {
    query: FETCH_API,
},
result: {
    data: {
        getUpcoming: "2",
        getPast: "109",
        getAll: "111"
    },
},
};

it('renders data', () => {
    const component = shallow(
      <MockedProvider addTypename={false} mocks={mocks}>
        <FeaturedInfo />
      </MockedProvider>,
    );

    expect(component).toBeTruthy();
    const up = component.children.findByType('div')
    expect(up).toContain('2')
});