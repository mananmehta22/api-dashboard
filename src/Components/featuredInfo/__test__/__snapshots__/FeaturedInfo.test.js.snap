// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render data 1`] = `
<div
  className="featured"
>
  <div
    className="featuredItem"
  >
    <span
      className="featuredTitle"
    />
    <div
      className="featuredLaunchContainer"
    >
      <span
        className="featuredUpcoming"
      >
        2
      </span>
    </div>
    <span
      className="featuredSub"
    >
      Upcoming
    </span>
  </div>
  <div
    className="featuredItem"
  >
    <span
      className="featuredTitle"
    />
    <div
      className="featuredLaunchContainer"
    >
      <span
        className="featuredUpcoming"
      >
        109
      </span>
    </div>
    <span
      className="featuredSub"
    >
      Past
    </span>
  </div>
  <div
    className="featuredItem"
  >
    <span
      className="featuredTitle"
    />
    <div
      className="featuredLaunchContainer"
    >
      <span
        className="featuredUpcoming"
      >
        111
      </span>
    </div>
    <span
      className="featuredSub"
    >
      Total
    </span>
  </div>
</div>
`;
